desktop/sources/scripts/client.js

```diff
@@ -26,6 +28,29 @@ function Client () {
   this.commander = new Commander(this)
   this.clock = new Clock(this)
 
+  // Ableton Link
+  this.link = new AbletonLink();
+
+  this.link.setTempoCallback((newTempo) => {
+    newTempo = this.link.getTempo(true)
+    if (this.clock.isLinkEnabled && this.clock.speed.value != newTempo) {
+      this.clock.setSpeed(newTempo, newTempo, true)
+      this.clock.setFrame(0)
```

Comment:
Won't this cause beat drift if Link tempo changes while playing?

I haven't implemented Ableton Link integration myself, but looking at their docs you probably want deeper integration to ensure Orca's clock adjusts appropriately and in sync.

Reply:
In my tests I haven't noticed any beat major drift, but as I said, it needs more testing.

I think the same about a deeper implementation, and that's the problem of using an external package and why this won't be merged into master.

I'll try to make some time to study the implementation and try to implement it from scratch 😄